src/guidellm/request/base.py

```diff
@@ -118,46 +129,31 @@ def create_item(self) -> TextGenerationRequest:
         """
         raise NotImplementedError()
 
-    def start(self):
-        """
-        Start the background task that populates the queue.
-        """
-        if self.mode == "async":
-            try:
-                loop = asyncio.get_running_loop()
-                logger.info("Using existing event loop")
-            except RuntimeError:
-                raise RuntimeError("No running event loop found for async mode")
-
-            loop.call_soon_threadsafe(
-                lambda: asyncio.create_task(self._populate_queue())
-            )
-            logger.info(
-                f"RequestGenerator started in async mode with queue size: "
-                f"{self._async_queue_size}"
-            )
-        else:
-            logger.info("RequestGenerator started in sync mode")
-
     def stop(self):
         """
         Stop the background task that populates the queue.
         """
         logger.info("Stopping RequestGenerator...")
         self._stop_event.set()
+        if self._mode == "async":
+            self._thread.join()
         logger.info("RequestGenerator stopped")
 
-    async def _populate_queue(self):
+    def _populate_queue(self):
         """
         Populate the request queue in the background.
         """
         while not self._stop_event.is_set():
-            if self._queue.qsize() < self._async_queue_size:
-                item = self.create_item()
-                await self._queue.put(item)
-                logger.debug(
-                    f"Item added to queue. Current queue size: {self._queue.qsize()}"
-                )
-            else:
-                await asyncio.sleep(0.1)
+            try:
+                if self._queue.qsize() < self._async_queue_size:
+                    item = self.create_item()
+                    self._queue.put(item, timeout=0.1)
+                    logger.debug(
+                        "Item added to queue. Current queue size: {}",
+                        self._queue.qsize(),
+                    )
+                else:
+                    time.sleep(0.1)
+            except Full:
+                continue
```

Comment:
We are going to use these constants for sleep and timeout?

Reply:
Good call, I'll migrate these to constants for general usage